shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/core/TokenizedNetworkClient.kt

```diff
@@ -0,0 +1,52 @@
+package co.nimblehq.blisskmmic.data.network.core
+
+import co.nimblehq.blisskmmic.BuildKonfig
+import co.nimblehq.blisskmmic.data.database.datasource.LocalDataSource
+import io.ktor.client.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.request.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.coroutines.flow.singleOrNull
+
+class TokenizedNetworkClient: NetworkClient {
+
+    val localDataSource: LocalDataSource
+
+    constructor(
+        engine: HttpClientEngine? = null,
+        localDataSource: LocalDataSource
+    ) : super(engine) {
+        this.localDataSource = localDataSource
+    }
+
+    override fun clientConfig(): HttpClientConfig<*>.() -> Unit {
+        return {
```

Comment:
```suggestion
        return super.clientConfig().apply{
           install(Auth) {
                bearer(bearerConfig())
            }
        }
```

Reply:
This is a nice suggestion but currently I'm not sure how the install should be declared. Maybe we can keep it as is.
<img width="1007" alt="Screen Shot 2022-12-13 at 11 56 42" src="https://user-images.githubusercontent.com/6356137/207230313-a1110145-c757-435c-8267-ec5b5de961ce.png">
<img width="576" alt="Screen Shot 2022-12-13 at 11 56 27" src="https://user-images.githubusercontent.com/6356137/207230329-d5a2dfed-0f6e-43a2-a801-a60258c83357.png">

Reply:
@blyscuit Agree, seem like the `install` function with `Auth` does not belongs to the HttpClientConfig.